documentation/architecture-decision-records/ADR0001-i18n_library.md

```diff
@@ -0,0 +1,102 @@
+
+The documentation is good https://github.com/yahoo/react-intl/wiki
+
+Support: 206 issues - 1 bug
+
+Licensing: Free to use under the Yahoo Inc. BSD license. See the LICENSE file for license text and copyright information.
+
+Examples:
+https://www.codeandweb.com/babeledit/tutorials/how-to-translate-your-react-app-with-react-intl
+https://medium.freecodecamp.org/internationalization-in-react-7264738274a0
+
+---
+
+1. `react-intl-native`
+
+Created and maintained by Alibaba.
+
+It is an improvement of `react intl` in that it adds functionality for supporting translations in Vanilla JS. React-intl only does this in React Components.
+
+Features:
+* It can do what we want i.e format numbers, times and dates
+* Can run in both the browser and Node js so we will be covered when we finally work on * ding SPA support )
+* Supports fetching of config files from a remote source e.g S3
+* Seems like it was built for a specific reason; supporting translations in Vanilla JS. As * ’ll be using React components, I don’t think we have to worry about this.
+
+Support:
+* Quick responses on both issues and Pull requests
+* 14 issues, 6 PRs
+
+* Developer experience:
+* Some parts of the docs aren’t grammatically correct. One will notice the errors but it’s not bad enough that one won’t be able to understand the docs.
+* Noticed that there are several issues that are asking for new features that are available on react-intl but not on react-intl-universal. Seems like they aren’t actively working on new features at the moment. If the current version suits our needs, we won’t have to worry about this.
+
+Licensing: BSD 3-Clause "New" or "Revised" License granted by Alibaba. License page info.
+
+---
+
+3. `fbt`
+
+Created by facebook.
+
+Features:
+* Organizing your source text for translation
+* Composing grammatically correct translatable UI
+* Eliminating verbose boilerplate for generating UI
+
+FBT works by transforming your `<fbt>` and `fbt(...)` constructs via Babel plugins. These plugins serve to extract strings from source and lookup translate payloads generated at build-time. FBT creates tables of all possible variations for the given fbt phrase and accesses this at runtime.
+
+Developer Experience:
+* Confusing syntax
+* Not many resources found, rather than the docs which are not very clear
+* https://facebookincubator.github.io/fbt/docs/api_intro
+* https://twitter.com/fbopensource/status/1080883562293878784
+
+Support: 18 issues - 1 bug. Low activity in community
+
+Licensing: FBT is MIT licensed, as found in the LICENSE file.
+
+---
+
+## Decision Outcome
+
+There was a clear preference among developers for option 1. `react-intl` as it is the most straight-forward and fully-supported option, providing the best developer experience out of all three. It was also the easiest to integrate into Psammead, as can be seen here: https://paper.dropbox.com/doc/2019-02-21-Component-Library-Meeting-i18n--AZgQeBdv~qXuQnL9jK5rcz4XAg-hSXxc23xnF6U7xnEkXwPc
```

Comment:
Can we also write an option for rolling our own please? 

Devils advocate, writing a function to do
```
<FormattedMessage
    defaultMessage={`Hello {name}, you have {unreadCount, number}`}
    values={{name: <b>{name}</b>, unreadCount}}
/>
```
*Could* be not that tricky and might save us 10kb on the client for SPA

Reply:
So is this something that we're going to seriously consider and have meetings to discuss or is this just to include in the document? I would argue that if we do have the intention of rolling out our own i18n library, that we should take into account our looming deadline.

Reply:
We should consider all possible options in an ADR, Theres a good chance someone can come to this document in 4 months time and go "why didnt they just make their own? would have been simple" How you consider it is up to you, but the document should say why you did/didnt go for it

Reply:
ADR amended 👍